resources/buildstock.rb

```diff
@@ -100,40 +111,28 @@ def get_option_name_from_sample_number(sample_value, dependency_values)
     key_s = hash_to_string(dependency_values)
     key_s_downcase = key_s.downcase
 
-    num_matches = @rows_keys_s.count(key_s_downcase)
-    if num_matches > 1
-      if key_s.size > 0
-        register_error("Multiple rows found in #{@filename.to_s} with dependencies: #{key_s.to_s}.", @runner)
-      else
-        register_error("Multiple rows found in #{@filename.to_s}.", @runner)
-      end
-    elsif num_matches == 0
+    rownum = @rows_keys_s[key_s_downcase]
+    if rownum.nil?
       if key_s.size > 0
         register_error("Could not determine appropriate option in #{@filename.to_s} for sample value #{sample_value.to_s} with dependencies: #{key_s.to_s}.", @runner)
       else
         register_error("Could not determine appropriate option in #{@filename.to_s} for sample value #{sample_value.to_s}.", @runner)
       end
     end
 
-    rownum = @rows_keys_s.index(key_s_downcase)
```

Comment:
Avoid the expensive `.count` and `.index` methods, which could be called up to a hundred thousand times for larger TSVs.